test/gtest/gtest_activation.cpp

```diff
@@ -0,0 +1,347 @@
+#include "test.hpp"
+#include <array>
+#include <iostream>
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <miopen/activ.hpp>
+#include <miopen/miopen.h>
+#include <miopen/stringutils.hpp>
+#include <miopen/tensor.hpp>
+#include <utility>
+#include <fusionHost.hpp>
+#include "verify.hpp"
+#include "gtest/gtest.h"
+
+struct ActivationConfig
+{
+    size_t N;
+    size_t C;
+    size_t H;
+    size_t W;
+    friend std::ostream& operator<<(std::ostream& outStream, const ActivationConfig& activConfig)
+    {
+        return outStream << "N:" << activConfig.N << " C:" << activConfig.C
+                         << " H:" << activConfig.H << " W:" << activConfig.W;
+    }
+};
+
+struct TestActivation : public ::testing::TestWithParam<
+                            std::tuple<miopenDataType_t, miopenActivationMode_t, ActivationConfig>>
+{
+protected:
+    template <typename T>
+    void setup_impl()
+    {
+        double alpha = 0.95;
+        double beta  = 2.3;
+        double gamma = 3.4;
+
+        input = tensor<T>{activ_config.N, activ_config.C, activ_config.H, activ_config.W};
+
+        std::get<tensor<T>>(input).generate(tensor_elem_gen_integer{17});
+        std::get<tensor<T>>(dinput_cpu) = std::get<tensor<T>>(input);
+        std::get<tensor<T>>(dinput_gpu) = std::get<tensor<T>>(input);
+
+        miopenCreateActivationDescriptor(&activ_desc);
+        miopenSetActivationDescriptor(activ_desc, activ_mode, alpha, beta, gamma);
+
+        std::size_t n, c, h, w;
+        auto&& handle        = get_handle();
+        std::tie(n, c, h, w) = miopen::tien<4>(std::get<tensor<T>>(input).desc.GetLengths());
+        EXPECT_EQ(n, activ_config.N) << "Config: " << activ_config;
+        EXPECT_EQ(c, activ_config.C) << "Config: " << activ_config;
+        EXPECT_EQ(h, activ_config.H) << "Config: " << activ_config;
+        EXPECT_EQ(w, activ_config.W) << "Config: " << activ_config;
+        size_t total_mem =
+            4 * std::get<tensor<T>>(input)
+                    .desc.GetNumBytes(); // estimate based on both forward and backward passes
+        size_t device_mem = handle.GetGlobalMemorySize();
+
+        ASSERT_LT(total_mem, device_mem) << "Tensor exceeds system memory size";
+
+        std::get<tensor<T>>(output_gpu) =
+            tensor<T>{static_cast<size_t>(n), // n from miopenGetConvolutionForwardOutputDim ?
+                      static_cast<size_t>(c),
+                      static_cast<size_t>(h),
+                      static_cast<size_t>(w)};
+        std::get<tensor<T>>(output_cpu_ref) =
+            tensor<T>{static_cast<size_t>(n), // n from miopenGetConvolutionForwardOutputDim ?
+                      static_cast<size_t>(c),
+                      static_cast<size_t>(h),
+                      static_cast<size_t>(w)};
+
+        std::fill(
+            std::get<tensor<T>>(output_gpu).begin(), std::get<tensor<T>>(output_gpu).end(), NULL);
```

Comment:
Instead of NULL its better to initialize with Nans

Reply:
> Instead of NULL its better to initialize with Nans

Initialized with std::numeric_limits<T>::quiet_NaN()